app/pages/lab-pages-editor/components/TasksPage.jsx

```diff
@@ -0,0 +1,113 @@
+            type="button"
+          >
+            Add a new Task +
+          </button>
+          {/* Dev observation: the <select> should have some label to indicate it's for choosing the starting task. */}
+          <select
+            aria-label="Choose starting test"
+            className="flex-item"
+          >
+            <option disabled={true}>Choose starting test</option>
+          </select>
+        </div>
+        <ul className="steps-list" aria-label="List of Pages/Steps">
+          {/* WARNING: this should be workflow.steps */}
+          {Object.entries(workflow.tasks).map(([taskKey, task]) => (
+            <StepItem
+              task={task}
+              taskKey={taskKey}
+            />
+          ))}
+        </ul>
+      </section>
+    </div>
+  );
+}
+
+// WARNING/TODO: this should be handling steps, not tasks
+function StepItem({ task, taskKey }) {
+  const stepKey = 'TODO';
+  if (!task || !taskKey) return null;
+
+  // TODO: use Panoptes Translations API.
+  // e.g. pull from workflow.strings['tasks.T0.instruction']
+  // Task.instruction/question isn't particularly standardised across different task types.
+  const text = task.instruction || task.question || '';
+
+  return (
+    <li className="step-item">
+      <div className="grab-handle flex-row spacing-bottom-XS">
+        <button aria-label={`Rearrange Page ${stepKey} upwards`} className="plain" type="button">
+          <span className="fa fa-caret-up" role="img" />
+        </button>
+        <GripIcon color="#A6A7A9" /> {/* TODO: add drag/drop functionality. Perhaps this needs to be wider, too. */}
+        <button aria-label={`Rearrange Page/Step ${stepKey} downwards`} className="plain" type="button">
+          <span className="fa fa-caret-down" role="img" />
+        </button>
+      </div>
+      <ul className="tasks-list" aria-label={`List of Tasks for Page/Step ${stepKey}`}>
+        <li className="task-item">
+          <div className="flex-row spacing-bottom-M">
+            <span className="task-key">{taskKey}</span>
+            <span className="task-icon">
+              {/* TODO: change icon and aria label depending on task type*/}
+              <span
+                aria-label="Task type: text"
+                className="fa fa fa-file-text-o fa-fw"
+                role="img"
+              />
+            </span>
+            <span className="task-text flex-item">{text}</span>
+            <button aria-label={`Delete Page/Step ${stepKey}`} className="plain" type="button">
+              <span className="fa fa-trash" role="img" />
+            </button>
+            <button aria-label={`Copy Page/Step ${stepKey}`} className="plain" type="button">
+              <span className="fa fa-copy" role="img" />
+            </button>
+            <button aria-label={`Edit Page/Step ${stepKey}`} className="plain" type="button">
+              <span className="fa fa-pencil" role="img" />
+            </button>
```

Comment:
```suggestion
            <button aria-label={`Delete Page/Step ${stepKey}`} className="plain" type="button">
              <span className="fa fa-trash" />
            </button>
            <button aria-label={`Copy Page/Step ${stepKey}`} className="plain" type="button">
              <span className="fa fa-copy" />
            </button>
            <button aria-label={`Edit Page/Step ${stepKey}`} className="plain" type="button">
              <span className="fa fa-pencil" />
            </button>
```

I think this might be an example of ['too much accessibility'](https://webaim.org/blog/overly-accessible/), where well-meaning developers add extra code that actually makes the page harder to use.

The [first rule of ARIA](https://www.w3.org/TR/using-aria/#rule1) is basically "don't use ARIA" so be sparing with roles and states. Add them when a task can't be performed without them, or without using native HTML elements and attributes.

Reply:
By the way, if a button is labelled with an image, eg. an SVG icon, then the appropriate pattern is to label the image:
```html
<button><img alt='This is the button label' src='./icon.svg'></button>
```